### PR TITLE
Remove ConfigureOptions sugar for types/instances

### PR DESCRIPTION
**Pull request by HaoK**
cc @divega

### CODE REVIEW

File: src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -24,72 +24,6 @@ public static IServiceCollection AddOptions(this IServiceCollection services)
...
-
-            var serviceTypes = FindIConfigureOptions(configureInstance.GetType());
-            foreach (var serviceType in serviceTypes)
-            {
-                services.AddSingleton(serviceType, configureInstance);
-            }
-            return services;
-        }
-
         public static IServiceCollection Configure<TOptions>(this IServiceCollection services, Action<TOptions> setupAction)
```
Comment (divega) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: `setupAction` is a very weird parameter name given there nothing called setup around this code any more. `configureAction`?

Comment (HaoK) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: Don't we typically call this configureOptions in other places?

Comment (divega) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: Do we? I am not aware.

Comment (HaoK) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: Well actually AddIdentity/AddMvc call this setupAction still

Comment (divega) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: We shouldn't :disappointed:

Comment (HaoK) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: But I typically name `Action<TOptions>` configureOptions, these days anyways... AddAuthentication has it this way...

Comment (divega) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: I am ok with making it consistently `configureOptions` if that is your preference.

Comment (HaoK) [inline src/Microsoft.Extensions.Options/OptionsServiceCollectionExtensions.cs]: Sure lets go with that

**Comment by HaoK** [comment]
0002d4ebc03bdd44270714748a84625f16fc2ebe

**Pull request closed without merging**